### DISCUSSION TITLE
Possibility to scan community posts

### DISCUSSION THREAD

**Discussion by Kurdakov** [ThioJoe/YT-Spammer-Purge#162] [Ideas]
I think google api allows somehow discriminate between videos and community posts (posts in community tab of channels), so in theory it is possible to scan community posts too, because spammers also post there.

**Comment by ThioJoe** [comment]
It is not possible. Believe me I've looked. YouTube's API does not allow getting comments from community posts at all, despite them being a feature for years. It's absurd. I did make a feature request for this on the Google issue tracker forum, and they acknowledged it and added it to the list of feature requests (but I think they do that for most), so hopefully they'll add it eventually: https://issuetracker.google.com/u/1/issues/206519108

**Reply by Kurdakov** [reply]
@ThioJoe seems I found a way. Each comment has unique id (and parent id for subcomments), now it looks like api allows to work with comments in community tab:  https://developers.google.com/youtube/v3/docs/comments (just no videoId in request), now the question, how to get ids of comments, if there is no API for that.
There is a project which allows to get ids of comments without API:
https://github.com/egbertbouman/youtube-comment-downloader
here https://github.com/egbertbouman/youtube-comment-downloader/blob/master/youtube_comment_downloader/downloader.py

I replaced
YOUTUBE_VIDEO_URL = 'https://www.youtube.com/watch?v={youtube_id}'
to 
YOUTUBE_VIDEO_URL = 'https://www.youtube.com/post/{youtube_id}'
also reused id option ( -y ) of that program.

It got comments with ids (the format of received messages could be adapted to be the code in your utility) from post in community tab.
Did not try yet on some channel (I do not have community tab yet, and on moderated channel I'm somewhat hesitant to experiment, need to find some spare spammer comment to try, but will try in near future).

p.s. possible additional benefit: if to use above mentioned approach to retrieve comments and there are no scammers - then quota won't be used (though it's slow but still not too much).

**Reply by ThioJoe** [reply]
Hmm you're right, I've been messing with it and it seems to work with some modifications

**Reply by rbeer** [reply]
On first glance, you can do essentially the same, starting at `https://youtube.com/c/{channel_id}/community`, to retrieve the post ids.

The whole regex stuff is necessary, because the initial batch of comments/posts comes with the HTML, since YouTube uses server-side rendering (SSR). So the data that comes from the `/youtubei/v1/browse` endpoint is in that first request's HTML response. If you trigger an infite scroll load and then look at the DevTool's network tab, you see the request made to `.../browse?key={INNERTUBE_API_KEY}`. Makes it a bit easier to understand the structure of the response.

**Reply by rbeer** [reply]
> p.s. possible additional benefit: if to use above mentioned approach to retrieve comments and there are no scammers - then quota won't be used (though it's slow but still not too much).

That's likely true. We used exactly this approach at [shadowban-eu](https://github.com/shadowban-eu) for about 4 years and with millions of requests a month and all from the same IP, without ever having any serious issues. The benefit is that you can read the data with a guest token, i.e. even when the quota is exceeded, you simply get a new token and the quota is reset.
Again, this is my experience with Twitter. YouTube might be different there, but sure worth a try.

**Reply by ThioJoe** [reply]
The latest 2.4.0 release now has the community post scanning in an experimental form. (Requires exact ID input, doesn't fetch community post owner's ID, therefore only ideal for scanning one's own posts)
https://github.com/ThioJoe/YT-Spammer-Purge/releases/tag/v2.4.0

**Reply by Kurdakov** [reply]
> On first glance, you can do essentially the same, starting at `https://youtube.com/c/{channel_id}/community`, to retrieve the post ids.
> 
good hint. ex source of https://www.youtube.com/c/ThioJoe/community  has all links to community posts in a form
{"webCommandMetadata":{"url":"/post/Ugxxxxxxxxxx .... ,
so with regex (I'm not really using regex much, but I found some regex expressions to extract url: pairs on the net) it is possible to build 'five recent community posts'

**Reply by rbeer** [reply]
You won't have to change much of the script Thio added, yesterday; the one you posted. It parses the whole config object into a dict, if my quick glance was correct.

You just have to find the path of the desired data in that dict.
